test/src/e2e_vm_tests/test_programs/should_fail/insufficient_type_info/test.toml

```diff
@@ -2,3 +2,6 @@ category = "fail"
 
 # check: foo()
 # nextln: $()Cannot infer type for type parameter "T". Insufficient type information provided. Try annotating its type.
+
+# check: Option::None::<T>;
+# nextln: $()Cannot infer type for type parameter "T". Insufficient type information provided. Try annotating its type.
```

Comment:
Sorry, I find this error message a bit confusing. I don't understand this part: `type for type parameter "T"`. Is sounds like we are trying to infer `T : some_type`, which is clearly not the case. I would expect the error message to say something like the type parameter `T` is unbound or not in scope.

Reply:
You are right, this is actually a secondary message.
The first error message is `Could not find symbol "T" in this scope.`. I will add that to the test.

Reply:
The PR looks good. I will approve after this comment is resolved.